### PR TITLE
Vmtest

### PR DESCRIPTION
**Pull request by Dinaamagdy**

### CODE REVIEW

**Comment by Dinaamagdy** [comment]
ARNING  - `j.clients.zero_os` is deprecated, please use `j.clients.zos` instead
ZRT-ZOS-003 ... ok
ZRT-ZOS-003 ... SKIP: https://github.com/zero-os/0-templates/issues/117
ZRT-ZOS-003 ... SKIP: https://github.com/zero-os/0-templates/issues/117

----------------------------------------------------------------------
Ran 3 tests in 10.467s

OK (SKIP=2)

File: tests/integration_tests/testsuite/a_basic/vms_tests.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+        *Test case for creating vm with non-valid parameters*
+
+        **Test Scenario:**
+        #. Create a vm without providing flist parameter, should fail.
+
+        """
+        self.log('Create a vm without providing flist parameter, should fail.')
+        vm1_name = self.random_string()
+        self.vms = {vm1_name: {     
+                               'nics': {"name": self.random_string(), "type": "default"}}}
```
Comment (john-kheir) [inline tests/integration_tests/testsuite/a_basic/vms_tests.py]: i believe u don't have to put nics

File: tests/integration_tests/testsuite/a_basic/vms_tests.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+class VM_actions(ZOS_BaseTest):
+    def __init__(self, *args, **kwargs):
+        super(VM_actions, self).__init__(*args, **kwargs)
+
+    @classmethod
+    def setUpClass(cls):
+        self = cls()
+        super(VM_actions, cls).setUpClass()
+        super(VM_actions, self).setUp()
```
Comment (john-kheir) [inline tests/integration_tests/testsuite/a_basic/vms_tests.py]: why this setUp() is here ? .. now we don't have different id for each testcase

File: tests/integration_tests/testsuite/a_basic/vms_tests.py [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+        self.wait_for_service_action_status(self.vm1_name, res[self.vm1_name]['resume'])        
+         
+        self.log('Check that [vm1] is runninng ')
+        vms = self.zos_client.kvm.list()
+        vm1 = [vm for vm in vms if vm['name'] == self.vm1_name]
+        self.assertEqual(vm1[0]['state'], "running")
+
+        self.log('%s ENDED' % self._testID)
+
+    @unittest.skip("https://github.com/zero-os/0-templates/issues/117")
```
Comment (john-kheir) [inline tests/integration_tests/testsuite/a_basic/vms_tests.py]: 117 is in verification state .. please check this testcase again

File: tests/integration_tests/testsuite/a_basic/vms_tests.py (hunk 1 above)
Comment (Dinaamagdy) [reply tests/integration_tests/testsuite/a_basic/vms_tests.py]: that's right , I just add default nic as example, but I will remove it and make it in separate testcase

**Comment by Dinaamagdy** [comment]
root@js9:~/zero-os/0-templates/tests/integration_tests# nosetests -v -s testsuite/a_basic/vms_tests.py  --tc-file=config.ini
[Wed23 09:49] - ZeroOSFactoryDeprecated.py:12  :j.clients.zero_os    - WARNING  - `j.clients.zero_os` is deprecated, please use `j.clients.zos` instead
ZRT-ZOS-003 ... ok
ZRT-ZOS-004 ... ok
ZRT-ZOS-005 ... ok
ZRT-ZOS-006 ... ok

----------------------------------------------------------------------
Ran 4 tests in 31.494s

OK
root@js9:~/zero-os/0-templates/tests/integration_tests#

File: tests/integration_tests/testsuite/a_basic/vms_tests.py (hunk 2 above)
Comment (Dinaamagdy) [reply tests/integration_tests/testsuite/a_basic/vms_tests.py]: without it I get error on log message and I need it

**Comment by Dinaamagdy** [comment]
root@js9:~/zero-os/0-templates/tests/integration_tests# nosetests -v -s testsuite/a_basic/vms_tests.py  --tc-file=config.ini
[Mon28 14:08] - ZeroOSFactoryDeprecated.py:12  :j.clients.zero_os    - WARNING  - `j.clients.zero_os` is deprecated, please use `j.clients.zos` instead
ZRT-ZOS-003 ... ok
ZRT-ZOS-004 ... ok
ZRT-ZOS-005 ... ok
ZRT-ZOS-006 ... ok

----------------------------------------------------------------------
Ran 4 tests in 34.489s

OK

**Pull request merged**